hpOneView/resources/resource.py

```diff
@@ -33,57 +33,892 @@
+
+        Returns:
+            A dict with the updated resource data.
+        """
+        uri = self.data['uri']
+
+        resource = deepcopy(self.data)
+        resource.update(data)
+
+        logger.debug('Update async (uri = %s, resource = %s)' %
+                     (uri, str(resource)))
+
+        self.data = self._helper.do_put(uri, resource, timeout, custom_headers)
+
+        return self
+
+    def get_by(self, field, value):
+        """Get the resource by passing a field and its value.
+
+        Note:
+            This function uses get_all passing a filter.The search is case-insensitive.
+
+        Args:
+            field: Field name to filter.
+            value: Value to filter.
+
+        Returns:
+            dict
+        """
+        if not field:
+            logger.exception(RESOURCE_CLIENT_INVALID_FIELD)
+            raise ValueError(RESOURCE_CLIENT_INVALID_FIELD)
+
+        filter = "\"{0}='{1}'\"".format(field, value)
+        results = self.get_all(filter=filter)
+
+        # Workaround when the OneView filter does not work, it will filter again
+        if "." not in field:
+            # This filter only work for the first level
+            results = [item for item in results if str(item.get(field, "")).lower() == value.lower()]
+
+        return results
+
+    def get_by_name(self, name):
+        """Retrieves a resource by its name.
+
+        Args:
+            name: Resource name.
+
+        Returns:
+            Resource object or None if resource does not exist.
+        """
+        result = self.get_by("name", name)
+
+        if result:
+            data = result[0]
+            new_resource = self.new(self._connection, data)
+        else:
+            new_resource = None
+
+        return new_resource
+
+    def get_by_uri(self, uri):
+        """Retrieves a resource by its URI
+
+        Args:
+            uri: URI of the resource
+
+        Returns:
+            Resource object
+        """
+        self._helper.validate_resource_uri(uri)
+        data = self._helper.do_get(uri)
+
+        if data:
+            new_resource = self.new(self._connection, data)
+        else:
+            new_resource = None
+
+        return new_resource
+
+    def _get_default_values(self):
+        """Gets the default values set for a resource"""
+
+        if self.DEFAULT_VALUES:
+            api_version = str(self._connection._apiVersion)
+            values = self.DEFAULT_VALUES.get(api_version, {}).copy()
+        else:
+            values = {}
+
+        return values
+
+    def _merge_default_values(self):
+        """Merge default values with resource data."""
+        values = self._get_default_values()
+        for key, value in values.items():
+            if not self.data.get(key):
+                self.data[key] = value
+
+    @classmethod
+    def new(cls, connection, data):
+        """Returns a new resource object"""
+        return cls(connection, data)
+
+
+class ResourceHelper(object):
```

Comment:
Why do we need to keep all the classes in a single file? Its better to separate these, correct?

Reply:
ResourceClient class will be removed when all the resources are updated with the new Resource class. All other classes are related to the Resource class and better to keep under one namespace.